angr/sim_state.py

```diff
@@ -296,19 +296,6 @@ def addr(self):
             return ip
         return self.solver.eval_one(self.regs._ip)
 
-    @property
-    def options(self):
-        return self._options
-
-    @options.setter
-    def options(self, v):
```

Comment:
this looks like it was a transitional thing that we never really started adding deprecation warnings for the other things. Are we ok with breaking options setting?

Reply:
I'm also not sure why we want to remove it instead of adding a deprecation warning. It was originally my fault that I did not add the warning btw.

Reply:
Can we have a property setter without having a property getter? `self.options` being dynamic takes an insane amount of time. I think we should just pull the bandaid on this one.

Reply:
A bigger problem is why we are calling `state.options` for so many times. The same applies to `state.arch`. I profiled the former at some point and felt we could move some options out of hot paths/code regions.